src/Lecture1.hs

```diff
@@ -123,4 +142,15 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Integer -> [Integer] -> String
+lowerAndGreater n list = go n (0 :: Integer) (0 :: Integer) list
+    where go n' lower greater [] = show n' ++ " is greater than " ++ show greater ++ " elements and lower than " ++ show lower ++ " elements"
+          go n' lower greater (x:rest) = let newLowerCount = lower + (lowerCount n' x)
+                                             newGreaterCount = (greater + (greaterCount n' x))
+                                          in go n' newLowerCount newGreaterCount $ rest
+          lowerCount n'' x = if (n'' < x)
+                               then 1
+                               else 0
+          greaterCount n'' x = if (n'' > x)
+                                 then 1
+                                 else 0
```

Comment:
Alternatively, you could use guards (and in the future even `compare` after I explain pattern-matching) to write this without helper functions. Something like this:

```haskell
go n' lower greater (x:rest)
    | x < n' = go n' (lower + 1) greater rest
    | x < n' = go n' lower (greater + 1) rest
    | otherwise = go n' lower greater rest
```